ext/node/polyfills/http.ts

```diff
@@ -1419,6 +1419,20 @@ export class ServerResponse extends NodeWritable {
     this.#headers[name] = value;
     return this;
   }
+  appendHeader(name: string, value: string | string[]) {
+    if (Array.isArray(value)) {
+      this.#hasNonStringHeaders = true;
+    }
+    if (this.#headers[name] === undefined) {
+      this.#headers[name] = value;
+    } else {
+      if (!Array.isArray(this.#headers[name])) {
+        this.#headers[name] = [this.#headers[name]];
+      }
+      this.#headers[name].push(value);
```

Comment:
This creates an array like `["a", "b", ["c", "d"]]` (nested headers arrays). That doesn't actually work.

Please check what Node does here (whether it creates a new array that concats the two arrays, or whether it does this and then handles nested arrays in the serialization).

Reply:
This should work fine. I added a test. 

For  `appendHeader("foo", ["bar"])`,  `headers[name]` becomes `["bar"]` which is serialized properly when we set `#hasNonStringHeaders = true` (See L1422).

Nested headers array like `["foo", ["bar"]]` errors.

Reply:
I understand that toString will add spaces. However that doesn't work for things like Set-Cookie, where appending headers by joining them together with , is invalid.

We need to make sure this matches node behavior

Reply:
Updated, similar to what Node is [doing](https://github.com/nodejs/node/blob/fbdfe9399cf6c660e67fd7d6ceabfb106e32d787/lib/_http_outgoing.js#L770-L782)